src/osvr/USBSerial/USBSerialDevInfo_MacOSX.h

```diff
@@ -87,15 +85,62 @@ namespace usbserial {
 
             return (vendor_matches && product_matches);
         }
-
-    } // end namespace
+    }
 
     std::vector<USBSerialDevice>
     getSerialDeviceList(boost::optional<uint16_t> vendorID,
                         boost::optional<uint16_t> productID) {
 
-        //FIXME: STUB
         std::vector<USBSerialDevice> devices;
+
+        io_iterator_t serialPortIterator;
+        io_object_t serialPortService;
+        // find all serial ports
+        if (findSerialPorts(&serialPortIterator) == true) {
+            // iterate over serial ports, getting vid and pid
+            while ((serialPortService = IOIteratorNext(serialPortIterator)) !=
+                   0) {
+                const CFNumberRef vidObj =
+                    static_cast<CFNumberRef>(IORegistryEntrySearchCFProperty(
+                        serialPortService, kIOServicePlane, CFSTR("idVendor"),
+                        NULL, kIORegistryIterateRecursively |
+                                  kIORegistryIterateParents));
+                const CFNumberRef pidObj =
+                    static_cast<CFNumberRef>(IORegistryEntrySearchCFProperty(
+                        serialPortService, kIOServicePlane, CFSTR("idProduct"),
+                        NULL, kIORegistryIterateRecursively |
+                                  kIORegistryIterateParents));
+                const CFStringRef bsdPathObj =
+                    static_cast<CFStringRef>(IORegistryEntryCreateCFProperty(
+                        serialPortService, CFSTR(kIOCalloutDeviceKey),
+                        kCFAllocatorDefault, 0));
+
+                if (!vidObj || !pidObj || !bsdPathObj) {
+                    continue;
+                }
+                // handle device
+                uint16_t vid, pid;
+                CFNumberGetValue(vidObj, kCFNumberSInt16Type, &vid);
+                CFNumberGetValue(pidObj, kCFNumberSInt16Type, &pid);
+
+                // convert the string object into a C-string
+                CFIndex bufferSize = CFStringGetMaximumSizeForEncoding(
+                                         CFStringGetLength(bsdPathObj),
+                                         kCFStringEncodingMacRoman) +
+                                     sizeof('\0');
+                std::vector<char> bsdPathBuf(bufferSize);
+                CFStringGetCString(bsdPathObj, &bsdPathBuf[0], bufferSize,
```

Comment:
`.data()` is C++11? The things I forget.  No, not in this or any internal library - if it builds on VS2013 and some sampling of linux and mac compilers :D. (public APIs are C, with a C++03 header-only wrapper using Boost, that at least on the client end will probably be replaced by c++11 header-only wrappers)